src/Tribe/Widget/Widget_Abstract.php

```diff
@@ -0,0 +1,415 @@
+
+		/**
+		 * Applies a filter to instance arguments based on the registration slug of the widget.
+		 *
+		 * @since TBD
+		 *
+		 * @param array<string,mixed> $arguments Current set of arguments.
+		 * @param static              $instance  Which instance of widget we are dealing with.
+		 */
+		$arguments = apply_filters( "tribe_widget_{$registration_slug}_arguments", $arguments, $this );
+
+		return $arguments;
+	}
+
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function get_argument( $index, $default = null ) {
+		$arguments = $this->get_arguments();
+		$argument  = Arr::get( $arguments, $index, $default );
+
+		return $this->filter_argument( $argument, $index, $default );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function filter_argument( $argument, $index, $default = null  ) {
+		/**
+		 * Applies a filter to a specific widget argument, catch all for all widgets.
+		 *
+		 * @since TBD
+		 *
+		 * @param mixed               $argument The argument.
+		 * @param string|int          $index    Which index we indent to fetch from the arguments.
+		 * @param array<string,mixed> $default  Default value if it doesn't exist.
+		 * @param static              $instance Which instance of widget we are dealing with.
+		 */
+		$argument = apply_filters( 'tribe_widget_argument', $argument, $index, $default, $this );
+
+		$registration_slug = $this->get_registration_slug();
+
+		/**
+		 * Applies a filter to a specific widget argument, to a particular registration slug.
+		 *
+		 * @since TBD
+		 *
+		 * @param mixed      $argument The argument value.
+		 * @param string|int $index    Which index we indent to fetch from the arguments.
+		 * @param array      $default  Default value if it doesn't exist.
+		 * @param static     $instance Which instance of widget we are dealing with.
+		 */
+		$argument = apply_filters( "tribe_widget_{$registration_slug}_argument", $argument, $index, $default, $this );
+
+		return $argument;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function get_default_arguments() {
+		return $this->filter_default_arguments( $this->default_arguments );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function filter_default_arguments( array $default_arguments = [] ) {
+		/**
+		 * Applies a filter to instance default arguments.
+		 *
+		 * @since TBD
+		 *
+		 * @param array<string,mixed>  $default_arguments Current set of default arguments.
+		 * @param static               $instance          Which instance of widget we are dealing with.
+		 */
+		$default_arguments = apply_filters( 'tribe_widget_default_arguments', $default_arguments, $this );
+
+		$registration_slug = $this->get_registration_slug();
+
+		/**
+		 * Applies a filter to instance default arguments based on the registration slug of the widget.
+		 *
+		 * @since TBD
+		 *
+		 * @param array<string,mixed>  $default_arguments Current set of default arguments.
+		 * @param static               $instance          Which instance of widget we are dealing with.
```

Comment:
```suggestion
		 * Applies a filter to default instance arguments based on the registration slug of the widget.
		 *
		 * @since TBD
		 *
		 * @param array<string,mixed>  $default_arguments Current set of default arguments.
		 * @param static               $instance          The widget instance we are dealing with.
```